### PR TITLE
Add full conversion script

### PR DESCRIPTION
**Pull request by alessandratrapani**
- [x] edit segmentation metadata
- [x] time alignment

### CODE REVIEW

**Comment by alessandratrapani** [comment]
stub [here](https://catalystneuro-processing.s3.us-east-2.amazonaws.com/reimer-arenkiel-lab-to-nwb/seb-134_ses-22.nwb)

File: src/reimer_arenkiel_lab_to_nwb/embargo2024/embargo2024_convert_session.py [hunk 1]
```diff
@@ -2,86 +2,105 @@
...
     data_dir_path = root_path / "Reimer-Arenkiel-CN-data-share"
     output_dir_path = root_path / "Reimer-Arenkiel-conversion_nwb/"
     stub_test = True
-    session_date = "2022-07-21_12-09-48"
-    session_id = "22"
-    subject_id = "134"
 
+    keys = get_session_keys()
+
+    start_time = time()
```
Comment (weiglszonja) [inline src/reimer_arenkiel_lab_to_nwb/embargo2024/embargo2024_convert_session.py]:
```suggestion
```

File: src/reimer_arenkiel_lab_to_nwb/embargo2024/embargo2024_convert_session.py [hunk 2]
```diff
@@ -2,86 +2,105 @@
...
         data_dir_path=data_dir_path,
         output_dir_path=output_dir_path,
-        session_date = session_date,
-        session_id=session_id,
-        subject_id=subject_id,
+        key=keys[4],
         stub_test=stub_test,
     )
+    end_time = time()
+    print(f"Running time:{end_time - start_time}")
```
Comment (weiglszonja) [inline src/reimer_arenkiel_lab_to_nwb/embargo2024/embargo2024_convert_session.py]:
```suggestion
```

File: src/reimer_arenkiel_lab_to_nwb/embargo2024/embargo2024_convert_session.py [hunk 3]
```diff
@@ -2,86 +2,105 @@
...
-from natsort import natsorted
-
+from time import time
+from tqdm import tqdm
 from neuroconv.utils import load_dict_from_file, dict_deep_update
+from neuroconv.tools.nwb_helpers import configure_and_write_nwbfile
 
 from reimer_arenkiel_lab_to_nwb.embargo2024 import Embargo2024NWBConverter
-
+from reimer_arenkiel_lab_to_nwb.dj_utils import *
```
Comment (weiglszonja) [inline src/reimer_arenkiel_lab_to_nwb/embargo2024/embargo2024_convert_session.py]: I would make this import explicitly show what functions are imported from dj_utils

File: src/reimer_arenkiel_lab_to_nwb/embargo2024/embargo2024_convert_session.py [hunk 4]
```diff
@@ -2,86 +2,105 @@
 
 from pathlib import Path
 from typing import Union
-from zoneinfo import ZoneInfo
-from natsort import natsorted
-
+from time import time
```
Comment (weiglszonja) [inline src/reimer_arenkiel_lab_to_nwb/embargo2024/embargo2024_convert_session.py]:
```suggestion
```

File: src/reimer_arenkiel_lab_to_nwb/dj_utils.py [hunk 5]
```diff
@@ -97,15 +93,13 @@ def add_subject(nwbfile: NWBFile, key: dict = None, verbose: bool = False) -> No
         date_of_birth=datetime.datetime.combine(subject_info['dob'], datetime.datetime.min.time()),
         sex=subject_info["sex"] if subject_info["sex"] in ("M", "F") else "U",
         description=subject_info["mouse_notes"] or "no notes",
-        species="Mus musculus",
+        species="Mus Musculus",
```
Comment (weiglszonja) [inline src/reimer_arenkiel_lab_to_nwb/dj_utils.py]: why make the second part uppercase? has there been a change in best practices?

File: src/reimer_arenkiel_lab_to_nwb/dj_utils.py [hunk 6]
```diff
@@ -193,73 +180,52 @@ def add_summary_images(nwbfile: NWBFile, key: dict = None, verbose: bool = False
...
+            indicator="GCaMP6f",
+            location="Whole Brain",
         ),
     )
 )
 
 
 def add_imaging_plane(
-    nwbfile: NWBFile, device: Device, key: dict = None, metadata: dict = None, verbose: bool = False
+        nwbfile: NWBFile, device: Device, key: dict = None, metadata: dict = None, verbose: bool = False
```
Comment (weiglszonja) [inline src/reimer_arenkiel_lab_to_nwb/dj_utils.py:205]: why not using `add_imaging_plane` from neuroconv?

File: src/reimer_arenkiel_lab_to_nwb/dj_utils.py (hunk 5 above)
Comment (weiglszonja) [reply src/reimer_arenkiel_lab_to_nwb/dj_utils.py]: yeah this should be "Mus musculus" as before (see inspector snippet below)

**Comment by weiglszonja** [comment]
@alessandratrapani can you fix these?

Screenshot 2024-07-22 at 15 54 29 (2)

File: src/reimer_arenkiel_lab_to_nwb/embargo2024/embargo2024_convert_session.py [hunk 7]
```diff
@@ -2,86 +2,105 @@
...
-    folder_path = data_dir_path / Path(session_date)
+    folder_path = data_dir_path / f"{key['animal_id']}_{key['session']}_{key['scan_idx']}"
 
     output_dir_path = Path(output_dir_path)
     if stub_test:
         output_dir_path = output_dir_path / "nwb_stub"
     output_dir_path.mkdir(parents=True, exist_ok=True)
 
-    nwbfile_path = output_dir_path / f"{subject_id}_{session_id}.nwb"
+    nwbfile_path = output_dir_path / f"seb-{key['animal_id']}_ses-{key['session']}.nwb"
```
Comment (weiglszonja) [inline src/reimer_arenkiel_lab_to_nwb/embargo2024/embargo2024_convert_session.py]: Probably a typo for subject:
```suggestion
    nwbfile_path = output_dir_path / f"sub-{key['animal_id']}_ses-{key['session']}.nwb"
```

File: src/reimer_arenkiel_lab_to_nwb/dj_utils.py [hunk 8]
```diff
@@ -344,11 +302,18 @@ def add_fluorescence(nwbfile, plane_segmentation, key: dict = None, verbose: boo
...
-def init_nwbfile(key: dict) -> NWBFile:
+def init_nwbfile(key: dict, metadata: dict = None) -> NWBFile:
     data = (all_sessions.Session & key).fetch1()
+    metadata = metadata or default_ophys_metadata
 
-    nwbfile_kwargs = deepcopy(default_ophys_metadata["NWBFile"])
+    nwbfile_kwargs = deepcopy(metadata["NWBFile"])
 
     nwbfile_kwargs.update(dict(
         session_description=f"animal {data['animal_id']} session {data['session']}",
```
Comment (weiglszonja) [inline src/reimer_arenkiel_lab_to_nwb/dj_utils.py]: Can you add something more descriptive here? E.g. this session contains two photon imaging ...

File: src/reimer_arenkiel_lab_to_nwb/dj_utils.py (hunk 5 above)
Comment (alessandratrapani) [reply src/reimer_arenkiel_lab_to_nwb/dj_utils.py]: Actually don't know why I changed it

File: src/reimer_arenkiel_lab_to_nwb/dj_utils.py (hunk 6 above)
Comment (alessandratrapani) [reply src/reimer_arenkiel_lab_to_nwb/dj_utils.py:205]: I am not sure, this fun was added by Ben.

File: src/reimer_arenkiel_lab_to_nwb/dj_utils.py (hunk 8 above)
Comment (alessandratrapani) [reply src/reimer_arenkiel_lab_to_nwb/dj_utils.py]: Updated in the yaml file

**Comment by alessandratrapani** [comment]
> @alessandratrapani can you fix these?
> 
> Screenshot 2024-07-22 at 15 54 29 (2)

I fixed the first three by updating the session and subject metadata.
For the last two I opened an issue (#6 ) because I found out the timestamps are NaNs --> I will resolve this in another PR

File: src/reimer_arenkiel_lab_to_nwb/dj_utils.py (hunk 5 above)
Comment (alessandratrapani) [reply src/reimer_arenkiel_lab_to_nwb/dj_utils.py]: I changed it back to the correct case

**Review by weiglszonja — APPROVED** [review]

**Pull request merged**